File: backend/internal/model/todo.go
package model

import "backend/internal/types"

type Todo struct {
	ID          int              `json:"id"`
	Title       string           `json:"title"`
	Description string           `json:"description"`
	Completed   bool             `json:"completed"`
	SprintID    *int             `json:"sprint_id"`
	CreatedAt   types.CustomTime `json:"created_at"`
	UpdatedAt   types.CustomTime `json:"updated_at"`
}

type TodoSearchRequest struct {
	Title       *string `json:"title"`       // 部分一致検索（任意）
	Description *string `json:"description"` // 部分一致検索（任意）
	Completed   *bool   `json:"completed"`   // 完了状態でフィルタ（任意）
	SprintID    *int    `json:"sprint_id"`   // スプリントIDでフィルタ（任意）
}

File: frontend/lib/types/todo.ts
/**
 * Todo型定義
 * GoバックエンドのTodo構造体に対応
 */
export type Todo = {
  id: string
  title: string
  description: string
  completed: boolean
  sprint_id: string | null
  created_at: string
  updated_at: string
}

/**
 * Todo作成時のデータ
 */
export type CreateTodoData = Pick<Todo, "title" | "description"> & {
  sprint_id?: string | null
}

/**
 * Todo更新時のデータ
 */
export type UpdateTodoData = Partial<CreateTodoData> & {
  completed?: boolean
}
